src/yunohost/backup.py

```diff
@@ -165,28 +255,39 @@ def _clean_tmp_dir(retcode=0):
                 else:
                     hooks_filtered.add(hook)
 
-        if not hooks or hooks_filtered:
-            logger.info(m18n.n('backup_running_hooks'))
-            ret = hook_callback('backup', hooks_filtered, args=[tmp_dir],
-                                env=env_var)
-            if ret['succeed']:
-                info['hooks'] = ret['succeed']
-
-                # Save relevant restoration hooks
-                tmp_hooks_dir = tmp_dir + '/hooks/restore'
-                filesystem.mkdir(tmp_hooks_dir, 0750, True, uid='admin')
-                for h in ret['succeed'].keys():
-                    try:
-                        i = hook_info('restore', h)
-                    except:
-                        logger.warning(m18n.n('restore_hook_unavailable',
-                                hook=h), exc_info=1)
-                    else:
-                        for f in i['hooks']:
-                            shutil.copy(f['path'], tmp_hooks_dir)
+        logger.info(m18n.n('backup_running_hooks'))
+
+        # Execute hooks
+        env_dict = self._get_env_var()
+        ret = hook_callback('backup', hooks_filtered, args=[self.collect_dir],
+                            env=env_dict, chdir=self.collect_dir)
+
+        if ret['succeed']:
```

Comment:
I don't know, it's an old piece of code. We should check the return of hook_calback